protocol/flows/src/flowcontext/transactions.rs

```diff
@@ -10,34 +10,62 @@ use kaspa_p2p_lib::{
 };
 use std::time::{Duration, Instant};
 
-const REBROADCAST_INTERVAL: Duration = Duration::from_secs(30);
+const CLEANING_TASK_INTERVAL: Duration = Duration::from_secs(10);
+const REBROADCAST_FREQUENCY: u64 = 3;
 const BROADCAST_INTERVAL: Duration = Duration::from_millis(500);
 pub(crate) const MAX_INV_PER_TX_INV_MSG: usize = 131_072;
 
 pub struct TransactionsSpread {
     hub: Hub,
-    last_rebroadcast_time: Instant,
+    last_cleaning_time: Instant,
+    cleaning_task_running: bool,
+    cleaning_count: u64,
     transaction_ids: ProcessQueue<TransactionId>,
     last_broadcast_time: Instant,
 }
 
 impl TransactionsSpread {
     pub fn new(hub: Hub) -> Self {
-        Self { hub, last_rebroadcast_time: Instant::now(), transaction_ids: ProcessQueue::new(), last_broadcast_time: Instant::now() }
+        Self {
+            hub,
+            last_cleaning_time: Instant::now(),
+            cleaning_task_running: false,
+            cleaning_count: 0,
+            transaction_ids: ProcessQueue::new(),
+            last_broadcast_time: Instant::now(),
+        }
     }
 
-    /// Returns true if the time for a rebroadcast of the mempool high priority transactions has come.
-    ///
-    /// If true, the instant of the call is registered as the last rebroadcast time.
-    pub fn should_rebroadcast_transactions(&mut self) -> bool {
-        let now = Instant::now();
-        if now - self.last_rebroadcast_time < REBROADCAST_INTERVAL {
+    /// Returns true if the time has come for running the task cleaning mempool transactions
+    /// and if so, mark the task as running.
+    pub fn should_run_cleaning_task(&mut self) -> bool {
+        if self.cleaning_task_running || Instant::now() < self.last_cleaning_time + CLEANING_TASK_INTERVAL {
             return false;
         }
-        self.last_rebroadcast_time = now;
+        // Keep the launching times aligned to exact intervals
```

Comment:
why do we need this alignment to exact intervals? this results in the weird while loop below for summing up time

Reply:
This is just me being a bit maniac with time precision and finding no better way to do calculations with `Instant`.

We can of course get back to the usual strategy of storing last time = now.

Reply:
I found a way to keep the alignment with a formula